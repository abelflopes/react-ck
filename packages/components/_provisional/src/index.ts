export * from "./image";

export * from "./avatar";

export * from "./dropdown";

export * from "./dock";

export * from "./breadcrumbs";

export * from "./scrollable-container";

export * from "./position-engine";

export * from "./top-bar";

export * from "./snackbar";

export * from "./overlay-spinner";

export * from "./menu";

export * from "./tooltip";

export * from "./popover";

export * from "./chat";

export * from "./file-uploader";

export * from "./attachment";

export * from "./snippet";
