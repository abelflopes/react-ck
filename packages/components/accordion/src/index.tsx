import React from "react";
import { List, type ListProps } from "@react-ck/list";
import { Collapse } from "@react-ck/collapse";
import styles from "./styles/index.module.scss";

export interface AccordionItem {
  header: NonNullable<React.ReactNode>;
  children: NonNullable<React.ReactNode>;
}

export interface AccordionProps extends Omit<ListProps, "items"> {
  items: AccordionItem[];
}

/**
 * An accordion is a vertically stacked list of interactive items. Each item can be "collapsed" with only a summary visible or “expanded” to show the full content for that item.
 * @param props - {@link AccordionProps}
 * @returns a React element
 */

export const Accordion = ({
  items,
  ...otherProps
}: Readonly<AccordionProps>): React.ReactElement => (
  <List
    {...otherProps}
    items={items.map(({ header, children }) => (
      <Collapse key={String(JSON.stringify(header))} header={header}>
        <div className={styles.content}>{children}</div>
      </Collapse>
    ))}
  />
);
