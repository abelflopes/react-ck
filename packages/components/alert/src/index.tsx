import React from "react";
import classNames from "classnames";
import styles from "./styles/index.module.scss";
import { Text } from "@react-ck/text";

export interface AlertProps extends React.HTMLAttributes<HTMLDivElement> {
  heading?: string;
  /** Specifies the visual style of the alert  */
  skin?: "neutral" | "primary" | "negative" | "average" | "positive";
}

// TODO: add dismiss

/**
 * Alert is a short message that provides contextual feedback in a prominent way.
 * @param props - {@link AlertProps}
 * @returns a React element
 */

export const Alert = ({
  heading,
  skin = "neutral",
  children,
  className,
  ...otherProps
}: Readonly<AlertProps>): React.ReactElement => (
  <div {...otherProps} className={classNames(styles.root, styles[skin], className)}>
    {heading && (
      <Text type="h4" as="p" variation="bold" margin={false} className={styles.heading}>
        {heading}
      </Text>
    )}
    {children}
  </div>
);
