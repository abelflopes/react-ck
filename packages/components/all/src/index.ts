export * from "./text";

export * from "./divider";

export * from "./banner";

export * from "./card";

export * from "./container";

export * from "./chip";

export * from "./collapse";

export * from "./list";

export * from "./overlay";
