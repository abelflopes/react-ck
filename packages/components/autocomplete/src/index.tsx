import React, { useEffect, useId, useMemo } from "react";
import styles from "./styles/index.module.scss";
import classNames from "classnames";
import { useFormFieldContext, type FormFieldProps } from "@react-ck/form-field";
import { AutocompleteOption } from "./AutocompleteOption";

interface AutocompleteProps extends React.InputHTMLAttributes<HTMLInputElement> {
  skin?: FormFieldProps["skin"];
}

/**
 * Autocomplete is a text input that proactively suggests options based on the userâ€™s initial input. The suggested options appear alongside the input in a panel.
 * @param props - {@link AutocompleteProps}
 * @returns a React element
 */

const Autocomplete = ({
  skin,
  id,
  className,
  children,
  ...props
}: Readonly<AutocompleteProps>): React.ReactElement => {
  const listId = useId();
  const formFieldContext = useFormFieldContext();

  const computedSkin = useMemo(
    () => formFieldContext?.skin ?? skin ?? "default",
    [formFieldContext?.skin, skin],
  );

  const computedId = useMemo(() => formFieldContext?.id ?? id, [formFieldContext?.id, id]);

  // Validate usage inside form field
  useEffect(() => {
    // Is not inside form field, skip
    if (formFieldContext === undefined) return;

    // Is inside form field
    if (skin)
      throw new Error("When using autocomplete inside form field, define skin on the form field");
    else if (id)
      throw new Error("When using autocomplete inside form field, define id on the form field");
  }, [formFieldContext, id, skin]);

  return (
    <>
      <input
        {...props}
        id={computedId}
        list={listId}
        className={classNames(
          styles.root,
          formFieldContext === undefined && styles.standalone,
          className,
          styles[`skin_${computedSkin}`],
        )}
      />
      <datalist id={listId}>{children}</datalist>
    </>
  );
};

Autocomplete.Option = AutocompleteOption;

export { Autocomplete, type AutocompleteProps };
