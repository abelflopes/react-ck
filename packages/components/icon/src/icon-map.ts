export type IconMap = Array<{
  import: string;
  module: string;
  export: string;
}>;

// https://react-icons.github.io/react-icons/

export const iconMap = [
  {
    import: "BsBell",
    module: "react-icons/bs",
    export: "IconBell",
  },
  {
    import: "BsBellFill",
    module: "react-icons/bs",
    export: "IconBellFill",
  },
  {
    import: "GrClose",
    module: "react-icons/gr",
    export: "IconClose",
  },
  {
    import: "TbSettings",
    module: "react-icons/tb",
    export: "IconCog",
  },
  {
    import: "TbFolder",
    module: "react-icons/tb",
    export: "IconFolder",
  },
  {
    import: "SiLinkedin",
    module: "react-icons/si",
    export: "IconLinkedin",
  },
  {
    import: "SiCodersrank",
    module: "react-icons/si",
    export: "IconCodersRank",
  },
  {
    import: "SiNpm",
    module: "react-icons/si",
    export: "IconNpm",
  },
  {
    import: "SiCodepen",
    module: "react-icons/si",
    export: "IconCodepen",
  },
  {
    import: "SiGithub",
    module: "react-icons/si",
    export: "IconGitHub",
  },
  {
    import: "CgSpinnerTwoAlt",
    module: "react-icons/cg",
    export: "IconSpinner",
  },
  {
    import: "CgChevronLeft",
    module: "react-icons/cg",
    export: "IconChevronLeft",
  },
  {
    import: "CgChevronRight",
    module: "react-icons/cg",
    export: "IconChevronRight",
  },
  {
    import: "CgChevronUp",
    module: "react-icons/cg",
    export: "IconChevronUp",
  },
  {
    import: "CgChevronDown",
    module: "react-icons/cg",
    export: "IconChevronDown",
  },
  {
    import: "CgTrash",
    module: "react-icons/cg",
    export: "IconTrash",
  },
  {
    import: "CgHome",
    module: "react-icons/cg",
    export: "IconHome",
  },
  {
    import: "FaRegCircleUser",
    module: "react-icons/fa6",
    export: "IconUserCircle",
  },
  {
    import: "HiOutlineLightningBolt",
    module: "react-icons/hi",
    export: "IconLightning",
  },
  {
    import: "GrDocumentText",
    module: "react-icons/gr",
    export: "IconDocument",
  },
  {
    import: "BsArrowRightCircleFill",
    module: "react-icons/bs",
    export: "ArrowRightCircleFill",
  },
  {
    import: "IoMdAttach",
    module: "react-icons/io",
    export: "Attach",
  },
];
