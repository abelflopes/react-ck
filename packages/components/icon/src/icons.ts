// Icons - https://react-icons.github.io/react-icons/icons?name=si
import { BsBellFill } from "react-icons/bs";
import { GrClose } from "react-icons/gr";

export default {
  "bell-filled": BsBellFill,
  "close": GrClose,
};
