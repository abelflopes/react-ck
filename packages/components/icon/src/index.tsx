import styles from "./styles/index.module.scss";
/// React
import React from "react";
// Icons
import { type IconType } from "react-icons";
import icons from "./icons";
import { useThemeContext } from "@react-ck/theme";
import classNames from "classnames";

export interface IconProps extends Omit<IconType, "size"> {
  skin?: "default" | "inverted";
  /** Specifies the name of the icon to be rendered */
  name: keyof typeof icons;
}

/**
 * Icon is a visual symbol that indicates the purpose of an interface element. Itâ€™s used to represent ideas, content types, and actions.
 * @param props - {@link IconProps}
 * @returns a React element
 */

export const Icon = ({
  skin = "default",
  name,
  ...otherProps
}: Readonly<IconProps>): React.ReactElement => {
  const themeContext = useThemeContext();

  const Icon: IconType = icons[name];

  return (
    <Icon
      {...otherProps}
      size={16}
      className={classNames(styles.root, {
        [`${styles[skin]}`]: skin !== "default" && skin !== "inverted",
        [`${styles.inverted}`]: themeContext.inverted || skin === "inverted",
      })}
    />
  );
};
