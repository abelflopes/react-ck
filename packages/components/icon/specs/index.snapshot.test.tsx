import React from "react";
import renderer from "react-test-renderer";
import { Icon } from "../src/index";

describe("snapshot Icon", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Icon>x</Icon>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
