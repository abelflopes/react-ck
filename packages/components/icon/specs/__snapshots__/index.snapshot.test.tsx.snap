// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Icon renders correctly 1`] = `
<span
  className="root size_text skin_default"
>
  x
</span>
`;
